### DISCUSSION TITLE
关于 useScrollLock 的请教

### DISCUSSION THREAD

**Discussion by 19Qingfeng** [youzan/vant#12116] [Q&A] [answered]
比较好奇 `useScrollLock` 这个移动端解决穿透滚动的问题，既然已经利用 `touchMove` 事件判断滚动到底部/顶部之后会阻止默认行为了，为什么还需要给 body 增加 `overflow:hidden`，有大佬清楚这个是为什么吗，求指教。

代码是看懂了，但是不知道这样是为什么解决什么样的问题，请教下代码中为什么要结合 `body{overflow:hidden}` 以及 `onTouchMove` 中的逻辑来进行双重计算。求指教，不甚感激

**Comment by chenjiahan** [comment, marked as answer]
移动端通常有 touchMove 的判断就够了，`overflow:hidden` 主要还是为了兼容一下 PC 端滚动的场景，以及预防移动端可能存在的边界情况。
